### ISSUE TITLE
Inconsistent context menu behavior

### ISSUE THREAD

**Issue by JeffBuckles** [mate-desktop/caja#1714]
#### Expected behaviour
Right mouse button (press, release) (ie "right-click") pops up the context menu, but does not select any menu item.   Menu remains open until any mouse button click.   
Example:  right click in left sidebar or on navigation tab label
(This preferred behavior is consisten with other desktop environments and tool menus)

#### Actual behaviour
Right click selects first item in context menu, and the menu is immediately closed.
Examples: right click on desktop or in navigation pane --> creates new empty folder

Note: right mouse button (press, drag, release) works as expected in both cases.

#### Steps to reproduce the behaviour
Right click (press, release without moving mouse) on the desktop or in the navigation pane

I notice a difference in the way the two menus are created:

The tab context menu, for example, calls "gtk_widget_show_all(popup)" before"gtk_menu_popup_at_pointer"  (Example: caja-navigation-window-pane.c:414),
while the desktop popup does not (Example: caja-information-panel.c:694, eel-gtk-extensions.c:294)

#### MATE general version
1.22.1

#### Package version
1.22.1-2.1.x86_64

#### Linux Distribution
SLES 12 SP5

#### Link to bugreport of your Distribution (requirement)
Not possible.  Corporate forces use of SLES 12, and SuSE bugreports are accepted only for SLES 15.   However, I see the conditions I believe to be causing this inconsistency present in the current version of Caja here in github.

Thanks and Best Regards,
--

**Comment by JeffBuckles** [comment]
Hi, lukefromdc, thanks for checking.
Do I understand correctly that if you right-click on the desktop, the context menu opens and stays open until you click something again?

We are running SLES 12 SP5 with the Mate desktop on virtual machines connected through ETX.  

However, there is strong evidence that this remote connection is not contributing to the issue.  If that were the reason for this problem, then I would expect _all_ right-click context menus to behave that way.   The problem occurs regardless of what input device I use (mouse, trackball, touchpad), and occurs on all of the linux desktop servers that I have access to, and regardless of which client I am connecting from.   It does _not_ occur in any of the engineering tools, or even in Caja when I right-click in the left (folder) pane or on tab labels.

I do not suspect hardware issues for the same reasons given above.   The behaviour is consistent across platforms, devices, and time.  That is, right click on tab labels or in the folder pane -- and in other tools --  always opens the context menu without selecting an item.   Only on the desktop and in the Caja navigation pane do I see it immediately select the first item and close the menu with a single right-click.   I am primarily concerned about the desktop problem.  

The problem began when we were forced to update from SLES 11 to SLES 12 about asix months ago, so it's been happening for a while, and I'm only now getting around to reporting it because I have time.

**Comment by lukefromdc** [comment]
Yes, on my setup the desktop context menu stays open until I click on something in it, or dismiss it by clicking on the desktop outside the menu area (the menu's popup window).
